assets/js/components/WebSocketApp.jsx

```diff
@@ -1,59 +1,65 @@
-        //
-        // socket.onmessage = (message) => {
-        //     console.log('hi');
-        // };
-
-        ws.onopen = function(e) {
-            console.log("Connection open...");
+        ws.onopen = function open(event) {
+            console.log('connected');
         };
 
-        // ws.onclose = function (e) {
-        //     console.log("Connection closed...");
-        // }
-
-        // client.onmessage = (message) => {
-        //     const dataFromServer = JSON.parse(message.data);
-        //     const stateToChange = {};
-        //     if (dataFromServer.type === "userevent") {
-        //         stateToChange.currentUsers = Object.values(dataFromServer.data.users);
-        //     } else if (dataFromServer.type === "contentchange") {
-        //         stateToChange.text = dataFromServer.data.editorContent || contentDefaultMessage;
-        //     }
-        //     stateToChange.userActivity = dataFromServer.data.userActivity;
-        //     this.setState({
-        //         ...stateToChange
-        //     });
-        // };
+        ws.onclose = function close() {
+            console.log('disconnected');
+        };
 
+        ws.onmessage = function incoming(message) {
+            // this.setState({newMessage: message})
+            console.log("message:" + message.data);
+        };
+
+        const b = fetch('http://127.0.0.1:8000/websocket/data', {method: 'GET'})
+            .then(response => response.json() )
+            .then(response =>
+                response.map(ab => this.setState({messages: [...this.state.messages, ab]}))
+            )
     }
 
     render() {
         return (
             <div>
                 Messages:
+                <ul>
+                    {this.state.messages.map(message =>
+                        <li key={message.timestamp}>{message.timestamp} - <strong>{message.username}</strong>: {message.message}</li>
+                    )}
+                </ul>
 
                 <div>
-                    <form onSubmit={event => socket.emit("message", "12345")}>
-                        <input type="text" name="name" onChange={event => this.setState({newMessage: event.target.value})} value={this.state.newMessage} />
+                    <form onSubmit={event => this.handleSubmit(event)}>
+                        <input type="text" id="text" name="name" onChange={event => this.setState({newMessage: event.target.value})} value={this.state.newMessage} />
                         <input type="submit" value="Submit"/>
                     </form>
                 </div>
             </div>
         );
     }
+
+    handleSubmit(event) {
+        fetch(
+            '/websocket/save',
+            {method: 'POST', body: httpBuildQuery({message: this.state.newMessage}), headers: {'Content-Type': 'application/x-www-form-urlencoded'}}
+        )
+        const text = document.getElementById('text').value
+        // console.log(this.state.newMessage)
+        ws.send(text)
```

Comment:
Hier sendest Du nur die reine Nachricht: woher wissen die anderen Clients von wem die Nachricht kam? timestamp könnte man auch hier direkt machen. Versuche die Nachricht möglichst nah in dem mercure Code zu halten.